Feature/search list data

baraa0abd opened the pull request:
feat: implement recent search data layer

This PR introduces the data layer components for managing recent searches:

- **RecentSearchEntity**: Defines the Room entity for storing individual recent search items.
- **RecentSearchResultEntity**: Defines the Room entity for storing search results.
- **RecentSearchDao**: Provides Data Access Object (DAO) methods for interacting with the `recent_searches` table in the Room database.
- **SearchResultDao**: Provides DAO methods for interacting with the `search_results` table.
- **InMemoryRecentSearchRepository**: An in-memory implementation of `RecentSearchRepository` for testing or non-persistent scenarios.
- **RoomRecentSearchRepository**: A Room-based implementation of `RecentSearchRepository`. The synchronous methods currently throw `UnsupportedOperationException` as they are intended to be replaced by suspend functions.

baraa0abd committed:
feat: Implement Search Screen and Recent Searches functionality

This commit introduces a new SearchScreen composable and a SearchViewModel to manage search-related state and logic.

Key features added:
- Search input field with clear and submit actions.
- Display of recent searches.
- Ability to add, remove, and clear recent search items.
- Filtering of recent searches based on current input.
- Basic loading and error states for recent searches.

New components:
- `SearchHeader`: Composable for the search input field and title.
- `RecentSearchList`: Composable to display the list of recent search queries.
- `RecentSearchItem`: Data class representing a single recent search.

Drawable resources for icons (clock and delete) have also been added.
Gradle configuration has been updated, including SDK versions and dependencies for Compose, ViewModel, LiveData, and Retrofit.
The `MainActivity` has been updated to display the `SearchScreen`.

baraa0abd committed:
chore: Update README.md with new content

baraa0abd committed:
refactor: Update RecentSearchList and SearchHeader to use AppTheme for styling

This commit refactors the RecentSearchList and SearchHeader components to utilize the AppTheme for consistent styling. Changes include:
- Replaced hardcoded padding, colors, and font styles with values from AppTheme.
- Improved layout and spacing for better visual consistency across the app.
- Enhanced readability by using theme-defined text styles and colors.

baraa0abd committed:
fix: update text styles and remove unused code

This commit updates the text styles in RecentSearchList and SearchHeader components to use the new text style definitions. It also removes unused code in MainActivity and RecentSearchList. Additionally, the project's JDK language level has been updated to 17, and the designSystem module has been added as a dependency.

baraa0abd committed:
chore: remove README.md file

baraa0abd committed:
Refactor: Use DesignSystem components in RecentSearchList and SearchHeader

This commit replaces standard Material Design components (Icon, IconButton, Text) with their DesignSystem equivalents (MovioIcon, MovioButton, MovioText) in the `RecentSearchList` and `SearchHeader` composables.

It also updates these components to use string resources for text and content descriptions, improving internationalization and accessibility.

baraa0abd committed:
Refactor: Integrate ViewModel with SearchScreen

This commit integrates `SearchListViewModel` into the `SearchScreen` composable.

Key changes:
- `SearchScreen` now instantiates and uses `SearchListViewModel`.
- `SearchListViewModel` manages the state of recent searches, including loading, adding, removing, and clearing them.
- `SearchScreen` observes the `recentSearches` flow from the ViewModel and updates the UI accordingly.
- Error handling for adding recent searches has been added.

baraa0abd committed:
feat: add RecentSearchItem data class and RecentSearchRepository interface

This commit introduces a data class `RecentSearchItem` to represent a recent search query with its ID, query string, timestamp, and an optional search count.

It also defines a `RecentSearchRepository` interface with methods for retrieving, adding, removing, and clearing recent search items.

baraa0abd committed:
feat: implement SearchListViewModel

This commit introduces `SearchListViewModel`, which manages the state and logic for recent searches.

It includes functionality to:
- Load recent searches
- Add a new recent search
- Remove a specific recent search
- Clear all recent searches

The ViewModel utilizes `InMemoryRecentSearchRepository` for data operations and exposes recent searches as a `StateFlow`.

baraa0abd committed:
refactor home and build gridle

baraa0abd committed:
delete uncomplete test file

Mohammed-qmr committed:
Add SearchScreen with ViewModel integration via Koin, search functionality, recommended & explore sections, and updated movie card with online image support

baraa0abd committed:
Merge branch 'develop' into feature/search-screen-ui

# Conflicts:
#	app/build.gradle.kts

baraa0abd committed:
Refactor: Use DesignSystem components in RecentSearchList and SearchHeader

This commit replaces standard Material Design components (Icon, IconButton, Text) with their DesignSystem equivalents (MovioIcon, MovioButton, MovioText) in the `RecentSearchList` and `SearchHeader` composables.

It also updates these components to use string resources for text and content descriptions, improving internationalization and accessibility.

Additionally, the `SearchHeader` component now uses `BasicTextField` instead of `TextField` for more customization and control over its appearance and behavior.

The `RecentSearchList` and `SearchHeader` composables have been moved from the `app` module to the `presentation` module, along with the `searchListScreen` composable and its associated ViewModel.

baraa0abd committed:
feat: implement recent search data layer

This commit introduces the data layer components for managing recent searches:

- **RecentSearchEntity**: Defines the Room entity for storing individual recent search items.
- **RecentSearchResultEntity**: Defines the Room entity for storing search results.
- **RecentSearchDao**: Provides Data Access Object (DAO) methods for interacting with the `recent_searches` table in the Room database.
- **SearchResultDao**: Provides DAO methods for interacting with the `search_results` table.
- **InMemoryRecentSearchRepository**: An in-memory implementation of `RecentSearchRepository` for testing or non-persistent scenarios.
- **RoomRecentSearchRepository**: A Room-based implementation of `RecentSearchRepository`. The synchronous methods currently throw `UnsupportedOperationException` as they are intended to be replaced by suspend functions.

fatimafaisalghazi commented on data/src/main/java/com/example/data/RecentSearchResultEntity.kt:
```diff
@@ -0,0 +1,13 @@
+    @PrimaryKey val queryKey: String, // query + language
+    val query: String,
+    val language: String,
+    val resultJson: String, // Store the result as JSON
```
delete comment

fatimafaisalghazi commented on presentation/src/main/java/com/example/presentation/component/searchUi/component/SearchHeader.kt:
```diff
@@ -0,0 +1,100 @@
+            ) {
+                MovioIcon(
+                    painter = painterResource(id = DesignSystemR.drawable.trash),
+                    contentDescription = stringResource(id = DesignSystemR.string.clear_content_description),
```
add it to string xml file

baraa0abd committed:
feat: Add RecentSearchItem and RecentSearchRepository

This commit introduces a new data class `RecentSearchItem` to represent a single recent search query and an interface `RecentSearchRepository` to define operations for managing recent search items.

The `RecentSearchItem` includes `id` and `query` properties.

The `RecentSearchRepository` interface defines methods for:
- Retrieving all recent searches
- Adding a new recent search item
- Removing a specific recent search item by its ID
- Clearing all recent searches

fatimafaisalghazi commented on app/src/main/java/com/example/movio/MainActivity.kt:
```diff
@@ -4,17 +4,16 @@ import android.os.Bundle
         setContent {
-            AppTheme (){
-
+            MovioTheme {
```
use AppTheme

baraa0abd committed:
style: update MainActivity theme to AppTheme

This commit updates the MainActivity to use `AppTheme` from the `designsystem` module instead of the local `MovioTheme`.

karrar-abbas commented:
resolve build error. it's about room library references
